tests/rptest/tests/read_distribution_test.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import bisect
+import time
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+from rptest.clients.kafka_cat import KafkaCat
+from rptest.services.redpanda import MetricsEndpoint
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class ReadDistributionTest(RedpandaTest):
+    def __init__(self, test_context):
+        super().__init__(test_context=test_context, num_brokers=1)
+
+        self.topic_name = 'tapioca'
+        self.topics = [TopicSpec(name=self.topic_name, replication_factor=1)]
+
+        self.record_size = 1024
+        self.record_count = 10
+        self.records_per_batch = 2
+        self.timespan_ms = 24 * 60 * 60 * 1000  #One day in ms.
+        self.timestamp_step_ms = self.timespan_ms // (self.record_count - 1)
+        self.curr_ts = round(time.time() * 1000)
+        self.base_ts = self.curr_ts - self.timespan_ms  #Start producing from record_count days ago.
+
+    @cluster(num_nodes=2)
+    def test_read_distribution_metric(self):
+        '''
+        Validate that Kafka reads are being correctly set in the read distribution histogram.
+        '''
+        KgoVerifierProducer.oneshot(
+            self.test_context,
+            self.redpanda,
+            self.topic_name,
+            self.record_size,
+            self.record_count,
+            batch_max_bytes=self.record_size * self.records_per_batch - 1,
+            fake_timestamp_ms=self.base_ts,
+            fake_timestamp_step_ms=self.timestamp_step_ms)
+        kcat = KafkaCat(self.redpanda)
+        for i in range(0, self.record_count):
+            timestamp = self.base_ts + i * self.timestamp_step_ms
+            kcat.consume_one(self.topic_name,
+                             0,
+                             offset=None,
+                             first_timestamp=timestamp)
+
+        metrics = self.redpanda.metrics(self.redpanda.nodes[0],
+                                        MetricsEndpoint.PUBLIC_METRICS)
+        read_dist_metric = [
+            m for m in metrics if m.name == 'redpanda_kafka_read_distribution'
+        ][0]
+        cumulative_topic_read_dist = [
+            (s.value, s.labels['le']) for s in read_dist_metric.samples if
+            s.labels['redpanda_topic'] == self.topic_name and 'le' in s.labels
+        ]
```

Comment:
nit: maybe worth checking that we have the right number of buckets in the histogram? Otherwise it seems like a bug could cause the distribution to be empty and we wouldn't notice

Reply:
Agreed!